### PR TITLE
fix: Return default collection ID in update collection API response

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
Return default collection id in the response of update action collection API

#### PR fixes following issue(s)
Fixes # [(ZH-68)](https://app.zenhub.com/workspaces/git-pod-63b7f52c55408d0011ef1e64/issues/zh/68)

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Improved the update logic for action collections to ensure more reliable data handling.
- **Tests**
	- Added tests for verifying the correct behavior of action collection updates.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/ResponseUtilsCE.java [hunk 1]
```diff
@@ -258,8 +258,14 @@ public ActionCollectionDTO updateCollectionDTOWithDefaultResources(ActionCollect
         collection.setId(defaultResourceIds.getCollectionId());
 
         // Update actions within the collection
-        collection.getActions().forEach(this::updateActionDTOWithDefaultResources);
-        collection.getArchivedActions().forEach(this::updateActionDTOWithDefaultResources);
+        collection.getActions().forEach(actionDto -> {
+            actionDto.setCollectionId(collection.getId());
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/ResponseUtilsCE.java:262]: Could you please help me understand why the value of `defaultResources.collectionId` at this point is incorrect?

Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/ResponseUtilsCE.java:262]: It's not incorrect, but both of them are same here. On line 258, we've `collection.setId(defaultResourceIds.getCollectionId())`. I understand that setting `defaultResourceIds.getCollectionId()` is more readable, will update that.

**Review by nidhi-nair — APPROVED** [review]

**Comment by nidhi-nair** [comment]
Ot seems like we never used to store default resources for actionDTO with collectionId and that is the root of the problem here. This PR just forcefully updates the references from the parent collection as a patch. The true fix is going to be to handle default resources in a clean way, which we have deferred to a complete project.

**Pull request merged**